src/Pos/Explorer/Web/Server.hs

```diff
@@ -135,9 +157,41 @@ searchHash shash = getResult $ do
         Left found -> return found
         Right _    -> throwM $ Internal "Search failed. No transactions, blocks or adresses found."
 
-    findTx      =  getTxSummary $ fromCSearchIdTx shash
-    findBlock   =  getBlockSummary $ fromCSearchIdHash shash
-    findAddress =  getAddressSummary $ fromCSearchIdAddress shash
+    findTx        =  getTxSummary $ fromCSearchIdTx shash
+    findBlock     =  getBlockSummary $ fromCSearchIdHash shash
+    findAddress   =  getAddressSummary $ fromCSearchIdAddress shash
+
+-- | Search the blocks by epoch and slot. Slot is optional.
+epochSlotSearch :: (ExplorerMode m) =>
+    EpochIndex ->
+    Maybe Word16 ->
+    m [CBlockEntry]
+epochSlotSearch epochIndex slotIndex = findBlocksByEpoch >>= transformToCBlockEntry
+  where
+    findBlocksByEpoch = getBlocksByEpoch @SscGodTossing epochIndex localSlotIndex
+    localSlotIndex = fmap LocalSlotIndex slotIndex
+    transformToCBlockEntry blocks = sequence $ toBlockEntry <$> blocks
+
+-- | Get all blocks by epoch and slot. The slot is optional, if it exists,
+-- it just adds another predicate to match it.
+getBlocksByEpoch :: (SscHelpersClass ssc, ExplorerMode m) =>
```

Comment:
I mean the rest of the codebase of `cardano-sl-explorer`, it seems `cardano-sl` is (mostly) written in that style.